lms/envs/common.py

```diff
@@ -2347,21 +2347,54 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
 
 ################################ Block Structures ###################################
 
+# .. setting_name: BLOCK_STRUCTURES_SETTINGS
+# .. setting_default: {
+#     COURSE_PUBLISH_TASK_DELAY: 30,
+#     TASK_DEFAULT_RETRY_DELAY: 30,
+#     TASK_MAX_RETRIES: 5,
+#     PRUNING_ACTIVE: False,
+#   }
+# .. setting_description: Stores all the settings used by block structures and block structure
+#   related tasks, such as updating the cached block structures.
+# .. setting_warnings: This setting is also used to specify the `STORAGE_CLASS` and
```

Comment:
https://github.com/edx/edx-platform/blob/ad5aa6cd7a6d61ca64521c92595fef85c5ba464c/openedx/core/djangoapps/content/block_structure/models.py#L62-L69

@robrap unfortunately, `BLOCK_STRUCTURES_SETTINGS['STORAGE_CLASS']` and `BLOCK_STRUCTURES_SETTINGS['STORAGE_KWARGS']` are directly accessed and used 😔 

I can move that information though, there?
Or maybe, throw an `ImproperlyConfigured` exception?

Reply:
I think we should either add the annotations where the setting is retrieved, or add new defaults that match the retrieval code and annotate. It may make sense to keep these annotations together? Thoughts?

Reply:
I do prefer adding new defaults and annotate 👍🏼 

However, this will need testing, to verify it. So, so I'll need to spin up two sandboxes to test two cases.
The first being with an S3 bucket, and the second being the file system.

It should be safe to specify something like, with the necessary annotations
```
STORAGE_CLASS = "django.core.files.storage.FileSystemStorage"
STORAGE_KWARGS = {}
DIRECTORY_PREFIX = "/tmp/courses/"
```

Should I move forward with this change?

Reply:
1. Maybe we should add the annotations where they are being retrieved for now to minimize risk?
2. If you want to follow up with a PR to move the annotations and set defaults, I’ll leave that to you.
3. Is the default for STORAGE_CLASS currently None? Is that a useless default that you are proposing to change, or did I miss this default somewhere?
3. Is DIRECTORY_PREFIX another one? Same applies around minimizing risk and just annotating somewhere that makes sense.

Let me know if this makes sense. Thanks.

Reply:
> Maybe we should add the annotations where they are being retrieved for now to minimize risk?

Alright, sure :+1: Sounds good. This change should be addressed now 😄 

> Is the default for STORAGE_CLASS currently None? Is that a useless default that you are proposing to change, or did I miss this default somewhere?

Yeah, the default for `STORAGE_CLASS` is currently `None`. And yeah, it is a useless default which I'm proposing to change. Let me know if you would prefer that I change the default value for it, though 👍🏼 

> Is DIRECTORY_PREFIX another one? Same applies around minimizing risk and just annotating somewhere that makes sense.

Yes, `DIRECTORY_PREFIX` is another one, which I forgot about 😆 
I'll annotate it similarly, as you requested 👍🏼